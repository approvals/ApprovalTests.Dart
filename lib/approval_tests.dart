/*
   Copyright 2024 shodev.live

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

// ignore_for_file: implementation_imports

import 'dart:convert';
import 'dart:io';
import 'dart:math';

import 'package:approval_tests/src/core/enums/file_type.dart';
import 'package:diff_match_patch2/diff_match_patch.dart';
import 'package:talker/talker.dart';
import 'package:test_api/src/backend/invoker.dart' show Invoker;

part 'src/approvals.dart';
part 'src/comparator/file_comparator.dart';
part 'src/core/approval_number.dart';
part 'src/core/approval_writer.dart';
part 'src/core/comparator.dart';
part 'src/core/enums/comporator_ide.dart';
part 'src/core/extensions/approval_string_extensions.dart';
part 'src/core/utils/file_path_extractor.dart';
part 'src/core/logger/logger.dart';
part 'src/core/options.dart';
part 'src/core/platform_wrapper.dart';
part 'src/core/reporter.dart';
part 'src/core/scrubber.dart';
part 'src/core/stack_trace_fetcher/fetcher.dart';
part 'src/core/stack_trace_fetcher/stack_trace_fetcher.dart';
part 'src/core/utils/converter.dart';
part 'src/core/utils/executable_query.dart';
part 'src/core/utils/utils.dart';
part 'src/exceptions/doesnt_match_exception.dart';
part 'src/exceptions/file_not_found_exception.dart';
part 'src/exceptions/no_diff_tool_exception.dart';
part 'src/namer/file_namer_options.dart';
part 'src/namer/namer.dart';
part 'src/reporters/command_line/command_line_reporter.dart';
part 'src/reporters/diff_tool/diff_info.dart';
part 'src/reporters/diff_tool/diff_tool_reporter.dart';
part 'src/reporters/diff_tool/diff_tools.dart';
part 'src/scrubbers/date_scrubber.dart';
part 'src/scrubbers/nothing_scrubber.dart';
part 'src/scrubbers/reg_exp_scrubber.dart';
part 'src/writers/approval_text_writer.dart';
part 'src/reporters/git.dart';
part 'src/core/constants/constants.dart';
