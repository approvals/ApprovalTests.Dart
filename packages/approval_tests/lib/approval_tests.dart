// ignore_for_file: implementation_imports

import 'dart:convert';
import 'dart:io';
import 'dart:math';

import 'package:diff_match_patch2/diff_match_patch.dart';
import 'package:talker/talker.dart';
import 'package:test_api/src/backend/invoker.dart' show Invoker;

part 'src/approvals.dart';
part 'src/comparator/file_comparator.dart';
part 'src/core/approval_number.dart';
part 'src/core/approval_writer.dart';
part 'src/core/comparator.dart';
part 'src/core/enums/comporator_ide.dart';
part 'src/core/extensions/approval_string_extensions.dart';
part 'src/core/utils/file_path_extractor.dart';
part 'src/core/logger/logger.dart';
part 'src/core/options.dart';
part 'src/core/platform_wrapper.dart';
part 'src/core/reporter.dart';
part 'src/core/scrubber.dart';
part 'src/core/stack_trace_fetcher/fetcher.dart';
part 'src/core/stack_trace_fetcher/stack_trace_fetcher.dart';
part 'src/core/utils/converter.dart';
part 'src/core/utils/executable_query.dart';
part 'src/core/utils/utils.dart';
part 'src/exceptions/doesnt_match_exception.dart';
part 'src/exceptions/file_not_found_exception.dart';
part 'src/exceptions/no_diff_tool_exception.dart';
part 'src/namer/file_namer_options.dart';
part 'src/namer/namer.dart';
part 'src/reporters/command_line/command_line_reporter.dart';
part 'src/reporters/diff_tool/diff_info.dart';
part 'src/reporters/diff_tool/diff_tool_reporter.dart';
part 'src/reporters/diff_tool/diff_tools.dart';
part 'src/scrubbers/date_scrubber.dart';
part 'src/scrubbers/nothing_scrubber.dart';
part 'src/scrubbers/reg_exp_scrubber.dart';
part 'src/writers/approval_text_writer.dart';
part 'src/reporters/git.dart';
