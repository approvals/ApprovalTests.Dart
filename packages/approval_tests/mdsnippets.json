{
  "$schema": "https://raw.githubusercontent.com/SimonCropp/MarkdownSnippets/master/schema.json",
  "Convention": "InPlaceOverwrite",
  "TocLevel": 5
}
