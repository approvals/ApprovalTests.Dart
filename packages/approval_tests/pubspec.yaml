#  * 1.0.0 for a first major release.
#  * 0.1.0 for a breaking release.
#  * 0.0.4 for a minor release.
name: approval_tests
description: Approval Tests implementation in Dart. Inspired by ApprovalTests.
version: 1.0.0
repository: https://github.com/approvals/ApprovalTests.Dart
homepage: https://github.com/approvals/ApprovalTests.Dart.git
issue_tracker: https://github.com/approvals/ApprovalTests.Dart/issues

topics:
  - test
  - approval
  - approvals
  - tests
  - approval-tests

environment:
    sdk: '>=3.0.0 <4.0.0'

dependencies:
  test: ^1.25.2
  talker: ^4.3.2
  diff_match_patch2: ^0.5.0
  test_api: ^0.7.0

dev_dependencies:
  sizzle_lints: ^2.0.2