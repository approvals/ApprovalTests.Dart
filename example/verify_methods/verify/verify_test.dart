import 'package:approval_tests/approval_tests.dart';
import 'package:test/test.dart';

void main() {
  test('Verify method', () {
    const String response =
        '{"result": "success", "data": {"id": 1, "name": "Item"}}';

    Approvals.verify(
      response,
      options: const Options(deleteReceivedFile: true),
    );
  });
}
