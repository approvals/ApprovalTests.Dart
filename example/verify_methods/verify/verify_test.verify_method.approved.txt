# This file was generated by approval_tests. Please do not edit.

{"result": "success", "data": {"id": 1, "name": "Item"}}